### PR TITLE
Add test for custom transliterator using Latin-ASCII and Lower

### PR DESCRIPTION
**Pull request by sffc**
It is a little awkward due to issues like #3991 and #3910, but I got it to work.

There are two other changes I split out into #5468 and #5467.

Depends on #5483

CC @FrankYFTang

### CODE REVIEW

File: components/experimental/tests/transliterate/lower_ascii.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                }) => (),
+                Err(e) => return Err(e),
+            }
+        }
+        self.0.load(req)
+    }
+}
+
+#[derive(Debug)]
+struct LowercaseTransliterator(CaseMapper);
```
Comment (Manishearth) [inline components/experimental/tests/transliterate/lower_ascii.rs:58]: suggestion: I suspect these should be exported by the transliterator crate

but we really need to figure out a better solution for bundling "default" transliterators anyway

File: components/experimental/tests/transliterate/lower_ascii.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        "<error>".to_string(),
+        ["Latin-ASCII"],
+        false,
+        true,
+    );
+    // Register Lower so that the alias mapping gets added
+    collection.register_source(
+        &"und-t-und-x0-lower".parse().unwrap(),
+        "<error>".to_string(),
+        ["Any-Lower"],
```
Comment (Manishearth) [inline components/experimental/tests/transliterate/lower_ascii.rs:82]: thought: would love transliterator to expose all of these in some module, perhaps with individual names, as well as a bundling solution that lets you include various useful bundles of the.

Comment (Manishearth) [reply components/experimental/tests/transliterate/lower_ascii.rs:82]:
```rust
trait NamedCustomTransliterator: CustomTransliterator {
    const BCP47: &'static str;
    const NAME: &'static str;
}
```

**Review by Manishearth — APPROVED** [review]

File: components/experimental/tests/transliterate/lower_ascii.rs (hunk 1 above)
Comment (sffc) [reply components/experimental/tests/transliterate/lower_ascii.rs:58]: Yes, they should be, but I didn't feel like making big changes right now; I want to show how to do this with minimal patches against 1.5

Comment (Manishearth) [reply components/experimental/tests/transliterate/lower_ascii.rs:58]: Yeah I'm not proposing that for this PR

**Comment by sffc** [comment]
I want #5483 to land first, but this PR should still be reviewed

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
I reverted the "Fix transliterator fallback loading" commit since it seems like I don't actually need it.

**Review by robertbastian — APPROVED** [review]

**Pull request merged**